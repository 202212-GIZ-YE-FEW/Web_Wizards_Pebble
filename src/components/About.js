const About = () => {
    return (
        <section id='about'>
            <div
                className='
                    relative 
                    overflow-hidden 
                    bg-no-repeat 
                    bg-cover 
                    w-full
                    brightness-200
                    h-[231px]
                    md:h-[493px]
                    lg:h-[690px]
                    xl:h-[787px]
                    overlay-container
                '
                style={{
                    backgroundPosition: "center",
                    backgroundImage: "url(/images/about.png)",
                }}
            ></div>
            <div
                className='
                    absolute
                    inset-0
                    opacity-60
                    text-center
                    lg:relative
                '
            >
                <h2
                    className='
                        font-black
                        top-1/2
                        text-xl
                        md:text-4xl
                        lg:text-5xl
                        text-dark
                        mt-16
                        mb-4
                        md:mt-52
                    '
                >
                    About Pebble Works
                </h2>
                <p
                    className='
                        text-sm
                        relative
                        mx-auto
                        w-1/2
                        md:text-xl
                        lg:text-3xl
                    '
                >
                    Body placeholder for text paragraph. A paragraph is a
                    self-contained unit of text dealing with a particular point
                    or idea.
                </p>
            </div>
        </section>
    );
};

export default About;
