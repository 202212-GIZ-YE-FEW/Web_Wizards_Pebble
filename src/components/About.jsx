function Brief(props) {
    const { head, brief } = props;
    return (
        <div className='absolute inset-0 mx-auto translate-y-1/4 h-full w-full flex-col items-center justify-center lg:relative'>
            <h2
                className='
                m-1 text-center text-xl mdP:text-3xl font-bold md:m-4 md:text-5xl md:font-bold lg:text-5xl lg:font-bold'
            >
                {head}
            </h2>
            <p
                className='
                text-sm mr-12 ml-12 text-center md:ml-24 md:mr-24 md:text-2xl lg:ml-32 lg:mr-32 lg:text-2xl'
            >
                {brief}
            </p>
        </div>
    );
}

const About = (props) => {
    const { t } = props;

    return (
        <section id='about'>
            <div
                className='relative overflow-hidden  bg-no-repeat  bg-cover w-full top-[-40px] brightness-200 h-[214px] sm:h-[489px] lg:h-[689px] xl:h-[789px] 2xl:h-[889px] before:absolute before:inset-0 before:bg-zinc-50 before:opacity-10'
                style={{
                    backgroundPosition: "center",
                    backgroundImage: "url(/images/about.png)",
                }}
            ></div>
            <Brief head={t("about")} brief={t("brief")} />
        </section>
    );
};

export default About;
