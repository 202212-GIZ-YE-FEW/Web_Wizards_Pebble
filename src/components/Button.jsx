import React from "react";

function Button({
    classes,
    children,
    handleClick = () => {},
    ariaLabel,
    dropDownButton = null,
}) {
    // console.log(classes);
    return (
        <>
            {dropDownButton === null ? (
                <button
                    onClick={() => handleClick(children)}
                    className={`button ${classes}`}
                >
                    {children}
                </button>
            ) : (
                <button
                    onClick={() => handleClick(children)}
                    className={`button ${classes}`}
                    aria-label={ariaLabel}
                    data-te-offcanvas-toggle
                    data-te-target={dropDownButton.target}
                    aria-controls={dropDownButton.controls}
                    data-te-ripple-init
                    data-te-ripple-color='dark'
                >
                    {children}
                </button>
            )}
        </>
    );
}

export default Button;
