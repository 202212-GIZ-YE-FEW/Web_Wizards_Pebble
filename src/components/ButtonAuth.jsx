import Image from "next/image";
import { Button } from "react-flatifycss";

import "flatifycss/dist/css/flatify-min.css";
import styles from "./ButtonAuth.module.css";

const ButtonAuth = (props) => {
    const { icon, alt, width, height } = props
    return (
        <>
            <Button
                className={`w-full h-14 px-6 transition ${styles.ButtonAuth}`}
            >
                <div className='relative flex items-center space-x-4 justify-center'>
                    <Image
                        src={icon}
                        alt={alt}
                        width={width}
                        height={height}
                    />
                    <span className='block w-max font-medium tracking-wide text-gray-900 sm:text-base'>
                        {props.text}
                    </span>
                </div>
            </Button>
        </>
    );
};

export default ButtonAuth;
