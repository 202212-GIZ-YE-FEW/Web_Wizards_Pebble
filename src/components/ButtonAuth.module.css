.ButtonAuth {
    background-color: white;
    border: 2px solid black;
    border-radius: 0.5rem;
    box-shadow: 2px 2px #1a1a1a;
}
