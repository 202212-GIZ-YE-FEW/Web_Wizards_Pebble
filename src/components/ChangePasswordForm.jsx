import EditprofileButton from "./EditProfileButton";
import InputForm from "./InputForm";
import SaveButton from "./SaveButton";
import SecondaryTitle from "./SecondaryTitle";
const ChangePasswordForm = ({ t }) => {
    return (
        <div className='w-11/12 mx-auto rounded-3xl flex flex-col justify-center bg-[#d4f3fa] md:px-10 px-2 py-10'>
            <SecondaryTitle text={t("changePassword")} />
            <div className='flex flex-row py-6'>
                <InputForm
                    placeholder={t("newPassword")}
                    label={t("newPassword")}
                    name='New Password'
                    type='Password'
                    hasFloatingLabel
                    togglePasswordLabel='Show/Hide password'
                    togglePassword='true'
                />
                <InputForm
                    placeholder={t("confirmPassword")}
                    label={t("confirmPassword")}
                    name='Confirm Password'
                    type='Password'
                    hasFloatingLabel
                    togglePasswordLabel='Show/Hide password'
                    togglePassword='true'
                />
            </div>

            <div className='flex flex-row '>
                <EditprofileButton text={t("cancel")} />
                <SaveButton text={t("submit")} />
            </div>
        </div>
    );
};

export default ChangePasswordForm;
