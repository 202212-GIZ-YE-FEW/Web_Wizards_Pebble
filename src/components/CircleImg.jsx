const CircleImg = () => {
    return (
        <div className='w-2/12 '>
            <div className='flex items-center justify-center w-32 h-32 md:w-36 md:h-36 lg:w-64 lg:h-64 rounded-full bg-black-100 text-white text-3xl font-bold'>
                <span className=''>R</span>
            </div>
        </div>
    );
};

export default CircleImg;
