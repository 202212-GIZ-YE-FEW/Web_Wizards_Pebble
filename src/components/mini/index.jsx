import { Button } from "react-flatifycss";

export function BorderlessButton(props) {
    const { children } = props;
    return (
        <Button className='!rounded-none !text-black-100 !bg-transparent !shadow-none after:content-[url("/arrow.svg")] after:mt-[1px] ltr:after:ml-4 after:absolute rtl:after:rotate-180 rtl:after:mr-4'>
            {children}
        </Button>
    );
}

export function Title(props) {
    const { className, children } = props;
    return (
        <h2
            className={`text-center text-black-100 text-4xl font-sans ${className}`}
        >
            {children}
        </h2>
    );
}
