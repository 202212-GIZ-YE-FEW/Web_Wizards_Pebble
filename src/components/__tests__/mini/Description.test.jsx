import renderer from "react-test-renderer";

import { Description } from "@/components/mini";

it("renders correctly", () => {
    const tree = renderer.create(<Description />).toJSON();
    expect(tree).toMatchSnapshot();
});
