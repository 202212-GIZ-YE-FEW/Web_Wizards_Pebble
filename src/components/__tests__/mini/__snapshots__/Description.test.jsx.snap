// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className="text-black-50 text-xl leading-8 "
/>
`;
